src/main/java/org/opensearch/sdk/BaseExtension.java

```diff
@@ -35,12 +36,37 @@ public abstract class BaseExtension implements Extension {
     protected ThreadPool threadPool;
 
     /**
-     * Empty constructor to fulfill abstract class requirements
+     * The extension settings include a name, host address, and port.
+     */
+    private ExtensionSettings settings;
```

Comment:
We are only ever setting this once, in the constructor, so it should be `final`.